src/Command/ConsumerCommand.php

```diff
@@ -0,0 +1,173 @@
+
+		if ($channel === null) {
+			throw new BunnyException("Could not open channel.");
+		}
+
+		$consumer = $this->consumers[$consumerName];
+
+		if ($consumer->getQueue() === null) {
+			$queueOk = $channel->queueDeclare("", false, false, true);
+
+			if (!($queueOk instanceof MethodQueueDeclareOkFrame)) {
+				throw new BunnyException("Could not declare anonymous queue.");
+			}
+
+			$consumer->setQueue($queueOk->queue);
+
+			$bindOk = $channel->queueBind($consumer->getQueue(), $consumer->getExchange(), $consumer->getRoutingKey());
+
+			if (!($bindOk instanceof MethodQueueBindOkFrame)) {
+				throw new BunnyException("Could not bind anonymous queue.");
+			}
+		}
+
+		if ($consumer->getPrefetchSize() || $consumer->getPrefetchCount()) {
+			$qosOk = $channel->qos($consumer->getPrefetchSize(), $consumer->getPrefetchCount());
+
+			if (!($qosOk instanceof MethodBasicQosOkFrame)) {
+				throw new BunnyException("Could not set prefetch-size/prefetch-count.");
+			}
+		}
+
+		if ($consumer instanceof BunnyTickingConsumerInterface && $consumer->getTickSeconds() === null) {
+			throw new BunnyException(
+				sprintf(
+					"If you implement 'TickerConsumerInterface', you have to specify 'tickSeconds' - %s.",
+					get_class($consumer)
+				)
+			);
+		}
+
+		$channel->consume(
+			function (Message $message, Channel $channel, Client $client) use ($consumer): void {
+				$this->handleMessage($consumer, $message, $channel, $client);
+			},
+			$consumer->getQueue(),
+			$consumer->getConsumerTag(),
+			$consumer->isNoLocal(),
+			$consumer->isNoAck(),
+			$consumer->isExclusive(),
+			false,
+			$consumer->getArguments()
+		);
+
+		$startTime = microtime(true);
+		$maxMessages = $consumer->getMaxMessages() ?: PHP_INT_MAX;
+		$maxSeconds = $consumer->getMaxSeconds() ?: PHP_INT_MAX;
+
+		while (microtime(true) < $startTime + $maxSeconds && $this->messages < $maxMessages) {
+			$channel->getClient()->run($consumer->getTickSeconds() ?: $maxSeconds);
+
+			if ($consumer instanceof BunnyTickingConsumer) {
+				$consumer->tick($channel);
+			}
+		}
+
+		$channel->getClient()->disconnect();
+	}
+
+	public function handleMessage(BunnyConsumerInterface $consumer, Message $message, Channel $channel, Client $client): void
+	{
+		$data = $message->content;
+
+		switch ((string) $message->getHeader("content-type")) {
+			case ContentTypes::APPLICATION_JSON:
+				$object = $this->serializer->deserialize($data, $consumer->getMessageClassName(), 'json');
+
+				break;
+			case ContentTypes::APPLICATION_PROTOBUF:
+				$object = $this->serializer->deserialize($data, $consumer->getMessageClassName(), ':protobuf');
+
+				break;
+			default:
+				throw new BunnyException("Message does not have 'content-type' header, cannot deserialize data.");
+		}
+
+		$consumer->handleMessage($object, $message, $channel, $client);
+		$maxMessages = $consumer->getMaxMessages();
+		$this->messages++;
+
+		if ($maxMessages !== null && $this->messages >= $maxMessages) {
```

Comment:
Není tohle duplicitní s kontrolou $maxMessages na ř.137? (možná ne, moc nevím, jak to spolu dohromady funguje)

Reply:
zcela uprimne nevim vypada to tak ale bojim se to odstranit protoze tahle to uz bylo a nahore se resi ze uz se nevola client->run a tady client->stop proc to tak je bych poresil v navaznem issue co ty na to ?

Reply:
Jasný, jestli to tak už bylo, tak bych to teď neměnil 👍